pyuvdata/mwa_beam.py

```diff
@@ -0,0 +1,573 @@
+# -- mode: python; coding: utf-8 --
+# Copyright (c) 2019 Radio Astronomy Software Group
+# Licensed under the 2-clause BSD License
+"""Read in the Sujinto et al. full embedded element MWA Beam."""
+from __future__ import absolute_import, division, print_function
+import numpy as np
+import warnings
+
+import h5py
+from scipy.special import factorial, lpmv  # associated Legendre function
+
+from . import UVBeam
+from . import utils as uvutils
+
+
+def P1sin(nmax, theta):
+    """
+    Create the Legendre function flavors for FF expansion using spherical waves.
+
+    Note this is not vectorized so is a bit slow, but it handles the special
+    case of theta = 0 and pi. We primarily use the vectorized version
+    (`P1sin_array`), but call this to handle the special cases.
+
+    See:
+    Calculating Far-Field Radiation Based on FEKO Spherical Wave Coefficients,
+    draft 10 June 2015.
+    Available at pyuvdata/docs/references/Far_field_spherical_FEKO_draft2.pdf
+    This memo gives a full description of the equations implemented here,
+    including descriptions of the approximations and numerical approaches used.
+    In line comments below are helpful reminders, but see the memo for the full
+    detail.
+    Also see Sokolowski, M. et al, "Calibration and Stokes Imaging with Full
+    Embedded Element Primary Beam Model for the Murchison Widefield Array",
+    PASA, 2017 (10.1017/pasa.2017.54) for details specific to the MWA.
+
+    Parameters
+    ----------
+    theta : float
+        (rad) is the cos_theta or sin_theta arguments
```

Comment:
This description isn't quite clear. Maybe `The argument of the cosine or sine function used in the associated Legendre functions, in radians.`?

Reply:
that's much better, thanks!